net/discovery.go

```diff
@@ -154,27 +156,46 @@ func (d *discovery) advertise() time.Duration {
 	return defaultAdvertiseTTL
 }
 
-func (d *discovery) discover(provides types.ProvidesCoin,
-	searchTime time.Duration) ([]peer.AddrInfo, error) {
-	log.Debugf("attempting to find DHT peers that provide [%s] for %vs...",
-		provides,
-		searchTime.Seconds(),
-	)
+func (d *discovery) discoverLoop() {
+	const discoverLoopDuration = time.Minute
+	timer := time.NewTicker(discoverLoopDuration)
+
+	for {
+		select {
+		case <-d.ctx.Done():
+			timer.Stop()
+			return
+		case <-timer.C:
+			if len(d.h.Network().Peers()) >= defaultMinPeers {
+				continue
+			}
+
+			// if our peer count is low, try to find some peers
+			timer := time.NewTimer(discoverLoopDuration)
```

Comment:
This is a shadowed variable that isn't even the same type as the outer scope version. I wonder why the linter didn't flag it.

Reply:
i think it's the same type, but yeah i'll rename it to avoid confusion

Reply:
The returned types are similar, but slightly different:
```
func NewTicker(d Duration) *Ticker
func NewTimer(d Duration) *Timer
```